README.md

```diff
@@ -1,11 +1,212 @@
 envoy-go-ldap-auth
 ==================
 
-This is a simple LDAP auth filter for envoy written in go.
+This is a simple LDAP auth filter for envoy written in go. Only requests that pass the LDAP server's authentication will be proxied to the upstream service.
 
-
-Status
-======
+## Status
 
 This is under active development and is not ready for production use.
 
+## Usage
+
+The client set credentials in `Authorization` header in the following format:
+
+```Plaintext
+credentials := Basic base64(username:password)
+```
+
+An example of the `Authorization` header is as follows (`aGFja2Vyczpkb2dvb2Q=`, which is the base64-encoded value of `hackers:dogood`):
+
+```Plaintext
+Authorization: Basic aGFja2Vyczpkb2dvb2Q=
+```
+
+Configure your envoy.yaml, include required fields: host, port, base_dn and attribute.
+
+```yaml
+http_filters:
+  - name: envoy.filters.http.golang
+    typed_config:
+      "@type": type.googleapis.com/envoy.extensions.filters.http.golang.v3alpha.Config
+      library_id: example
+      library_path: /etc/envoy/libgolang.so
+      plugin_name: envoy-go-ldap-auth
+      plugin_config:
+        "@type": type.googleapis.com/xds.type.v3.TypedStruct
+        value:
+          # required
+          host: localhost
+          port: 389
+          base_dn: dc=example,dc=com
+          attribute: cn
+          # optional
+          # be used in search mode
+          bind_dn: 
+          bind_password: 
+          # if the filter is set, the filter application will run in search mode.
+          filter: 
```

Comment:
better add an example config as demo, in comment could be good enough.

Reply:
OK